### PR TITLE
Create filterNot.md #3726

### PR DESCRIPTION
**Pull request by AlstrupJacob**
### Description

Closes issue 'Kotlin Strings .filterNot() #3726'

### Type of Change

created filterNot.md. entry in directory docs/content/kotlin/concepts/strings/terms/filterNot

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by AlstrupJacob** [comment]
should pass now

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+## Syntax
+
+```pseudo
+inline fun String.filterNot(
+    predicate: (Char) -> Boolean
+): String
+```
+
+`String`: The collection of elements to be filtered. can be any iterable dtype.
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:27]: The documentation shows that this available for Strings and CharSequence

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+```pseudo
+inline fun String.filterNot(
+    predicate: (Char) -> Boolean
+): String
+```
+
+`String`: The collection of elements to be filtered. can be any iterable dtype. 
+
+`Predicate`: The collection of elements to be filtered out.
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:29]:
```suggestion
`predicate`: The collection of elements to be filtered out.
```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+The `.filterNot(predicate)` method returns the collection it was called on, with any values present in `predicate` filtered out
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:31]:
```suggestion
```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## Example
+
+```kotlin
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:35]:
```suggestion
```kotlin
fun main {
```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+val originalMap = mapOf("a" to 1, "b" to 2, "c" to 3)
+
+val filteredMap = originalMap.filterNot { it.value < 2 }
+println(filteredMap) // {b=2, c=3}
+
+
+val filteredMap2 = originalMap.filterNot { it.key == 'c'
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:42]:
```suggestion
val filteredMap2 = originalMap.filterNot { it.key == "c" }
```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+}
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:43]:
```suggestion

```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+println(filteredMap2) // {a=1, b=2}
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:44]:
```suggestion
println(filteredMap2) 
}
```

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+The `.filterNot(predicate)` method returns the collection it was called on, with any values present in `predicate` filtered out
+
+## Example
+
+```kotlin
+val originalMap = mapOf("a" to 1, "b" to 2, "c" to 3)
+
+val filteredMap = originalMap.filterNot { it.value < 2 }
+println(filteredMap) // {b=2, c=3}
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:39]:
```suggestion
println(filteredMap)
```

Comment (caupolicandiaz) [reply content/kotlin/concepts/strings/terms/filterNot/filterNot.md:39]: the lines of the body should be indented 2 spaces

File: content/kotlin/concepts/strings/terms/filterNot/filterNot.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+
+val filteredMap2 = originalMap.filterNot { it.key == 'c'
+}
+println(filteredMap2) // {a=1, b=2}
+```
+
+In the above example .filterNot() is first used to filter out any pairs in the map with a value less than 2, leaving {b=2, c=3}, then used to filter out any pairs with the key 'c', leaving
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/strings/terms/filterNot/filterNot.md:47]:
```suggestion
The example demonstrates the use of `.filterNot()` to remove any pairs in the map with a value less than `2`, and then used to filter out any pairs with the key `"c:`.
```

Comment (caupolicandiaz) [reply content/kotlin/concepts/strings/terms/filterNot/filterNot.md:47]: This should come after the example header and before the code block

Comment (caupolicandiaz) [reply content/kotlin/concepts/strings/terms/filterNot/filterNot.md:47]: The output should be in a ```shell block

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@AlstrupJacob I've added my comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by AlstrupJacob** [comment]
@caupolicandiaz done

**Comment by AlstrupJacob** [comment]
@caupolicandiaz sorry for the bit of confusion this is my first contrib, all changes complete now

**Review by caupolicandiaz — APPROVED** [review]
@AlstrupJacob Just made a couple of direct edits to update the spacing. This is ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@AlstrupJacob Looks good.

**Pull request merged**